// custom_modules/email_mngr.js
'use strict';

const 
    nodemailer = require('nodemailer'),
    transporter = nodemailer.createTransport({direct:true,
        host: 'smtp.mail.ru',
        port: 465,
        auth: { 
            user: 'pryanik_bot@mail.ru', 
            pass: 'botAccount' },
        secure: true
    }),
    mailOptions = {
        from: 'pryanik_bot@mail.ru',
        to: 'more-pr@mail.ru',
        subject: '',
        text: ''
    };





    
let rand = function(min, max){
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

let createEmail = function(title, message)
{
    let uid = rand(100000, 999999);
    mailOptions.subject = `${title} || ${uid} ||`;
    mailOptions.text = message;
    transporter.sendMail(mailOptions, function(error, info){
        if (error) {
          console.log(error);
        } else {
          console.log('Email sent: ' + info.response);
        }
      });
}

exports.sendOrder = (name, msg, tel) => {
    createEmail(`Новый заказ от ${name}`,
                `Заказ от ${name} \n ${msg} \n\n \tТелефон для связи:${tel}`);
}

exports.sendFeedback = (name, email, tel, msg) => {
    createEmail(`Обратная связь от ${name}`,
                `Обратная связь от ${name} \n ${msg} \n\n \n\tE-mail: ${email}\n\tТелефон для связи:${tel}`);
}

exports.sendQuestion = (name, email, msg) => {
    createEmail(`Вопрос от ${name}`,
                `Вопрос от ${name} \n ${msg} \n\n \n\tE-mail: ${email}`);
}
